### PR TITLE
Adding genre information

### PR DESCRIPTION
**Pull request by tukkek**
See #366 for more information.

This is building without errors as per the `make` command.

There's a minor change to `_ext.py` that can be ignored if it isn't appropriate but I found it to be helpful.

### CODE REVIEW

File: games/a.yaml [hunk 1]
```diff
@@ -225,6 +225,9 @@
     - http://cyxdown.free.fr/rawgl/rawgl-2.png
 
 - name: ['Another World 2: Heart of the Alien', Heart of the Alien]
+  meta:
+    genre: [adventure, platform]
+  remakes:
```
Comment (cxong) [inline games/a.yaml]: redundant line

File: games/a.yaml [hunk 2]
```diff
@@ -234,6 +237,9 @@
     lang: C
 
 - name: 'Archon: The Light and the Dark'
+  meta:
+    genre: [TBS]
+  remakes:
```
Comment (cxong) [inline games/a.yaml]: redundant line

File: games/a.yaml [hunk 3]
```diff
@@ -319,6 +329,8 @@
     - http://daid.github.io/EmptyEpsilon/screenshots/helms.png
 
 - name: Artillery Duel
+  meta:
+    genre: [arcade]
```
Comment (cxong) [inline games/a.yaml]: I think this would be an artillery game (subgenre of strategy)

File: games/a.yaml [hunk 4]
```diff
@@ -557,6 +595,8 @@
     license: [MIT]
 
 - name: 'Artemis: Spaceship Bridge Simulator'
```
Comment (cxong) [inline games/a.yaml]: Looks like this game is repeated

File: games/b.yaml [hunk 5]
```diff
@@ -26,6 +30,8 @@
     - http://sourceforge.net/dbimage.php?id=105344
 
 - name: Ballerburg
+  meta:
+    genre: [arcade]
```
Comment (cxong) [inline games/b.yaml]: another artillery game

File: games/e.yaml [hunk 6]
```diff
@@ -1,4 +1,6 @@
 - name: [E.T. the Extra-Terrestrial, E.T. the Extra-Terrestrial (video game)]
+  meta:
+    genre: [arcade]
```
Comment (cxong) [inline games/e.yaml]: This should be an adventure game

File: games/f.yaml [hunk 7]
```diff
@@ -123,6 +137,8 @@
     framework: Construct
 
 - name: Flappy Bird
+  meta:
+    genre: [platform]
```
Comment (cxong) [inline games/f.yaml]: Should be arcade

File: games/g.yaml [hunk 8]
```diff
@@ -116,6 +125,8 @@
     - http://media.moddb.com/images/mods/1/5/4506/ro3.jpg
 
 - name: [Gorillas, Gorillas (video game)]
+  meta:
+    genre: [arcade]
```
Comment (cxong) [inline games/g.yaml]: artillery

File: games/l.yaml [hunk 9]
```diff
@@ -55,6 +63,8 @@
     lang: JavaScript
 
 - name: Liero
+  meta:
+    genre: [RTS]
```
Comment (cxong) [inline games/l.yaml]: should be platformer/shmup

File: games/m.yaml [hunk 10]
```diff
@@ -246,6 +277,8 @@
     - http://www.maxr.org/images/screenshots/10.jpg
 
 - name: [Mech Warrior, MechWarrior (video game series)]
+  meta:
+    genre: [FPS]
```
Comment (cxong) [inline games/m.yaml]: I think simulation would be a better fit here

File: games/s.yaml [hunk 11]
```diff
@@ -1,4 +1,6 @@
 - name: [Scorched Earth, Scorched Earth (video game)]
+  meta:
+    genre: [arcade]
```
Comment (cxong) [inline games/s.yaml]: artillery

File: games/s.yaml [hunk 12]
```diff
@@ -45,6 +55,8 @@
     license: [MIT]
 
 - name: Skool Daze
+  meta:
+    genre: [arcade]
```
Comment (cxong) [inline games/s.yaml]: not really arcade; maybe action adventure?

File: games/s.yaml [hunk 13]
```diff
@@ -70,6 +84,8 @@
     lang: [C++, Python]
 
 - name: Strike Commander
+  meta:
+    genre: []
```
Comment (cxong) [inline games/s.yaml]: simulation

File: games/s.yaml [hunk 14]
```diff
@@ -79,6 +95,8 @@
     updated: 2014-01-05
 
 - name: Strikers 1945
+  meta:
+    genre: [simulation]
```
Comment (cxong) [inline games/s.yaml]: shmup

File: games/s.yaml [hunk 15]
```diff
@@ -177,6 +207,8 @@
     license: [AGPL3]
 
 - name: Sid Meier's Pirates!
+  meta:
+    genre: [action,adventure,RTS]
```
Comment (cxong) [inline games/s.yaml]: add TBS, remove RTS - I don't think there are any RTS segments in the game

File: games/s.yaml [hunk 16]
```diff
@@ -870,6 +980,8 @@
     license: [GPL3]
 
 - name: [Snake, Snake (video game)]
```
Comment (cxong) [inline games/s.yaml]: Duplicate game

File: games/s.yaml [hunk 17]
```diff
@@ -800,6 +908,8 @@
     license: [GPL3]
 
 - name: [Similar to Natural Selection, Natural Selection (video game)]
```
Comment (cxong) [inline games/s.yaml]: This should be in the n.yaml file I think; not sure why "Similar to" is necessary

File: games/s.yaml [hunk 18]
```diff
@@ -883,6 +995,8 @@
     - https://a.fsdn.com/con/app/proj/gustysserpents/screenshots/218296.jpg
 
 - name: Sokoban
+  meta:
+    genre: []
```
Comment (cxong) [inline games/s.yaml]: puzzle

File: games/s.yaml [hunk 19]
```diff
@@ -978,6 +1098,8 @@
     - https://github.com/jazzyjester/Mario-Objects/raw/master/mario1.png
 
 - name: [Scorched Earth, Scorched Earth (video game)]
```
Comment (cxong) [inline games/s.yaml]: duplicate game

File: games/t.yaml [hunk 20]
```diff
@@ -771,6 +850,8 @@
     updated: 2013-05-28
 
 - name: Tetris
```
Comment (cxong) [inline games/t.yaml]: Duplicate

File: games/u.yaml [hunk 21]
```diff
@@ -107,6 +127,8 @@
     lang: JavaScript
 
 - name: UGH!
```
Comment (cxong) [inline games/u.yaml]: Duplicate?

**Comment by tukkek** [comment]
Oh dear, was writing a comment but GH lost it, let me try again :/

First: thanks for the approval and in-depth review :) do I need to do something on my end for the commit to go through?

As for the duplicates, I think some of the games have separate entries listing `clones` and `remakes`. Makes sense to me why there'd be different fields for that but I'm not sure why they aren't being done on the same game entry. Since I'm a newbie here I didn't want to change anything. Of course, as you can see from the commits these aren't my changes to the code, I only changed the "meta" tags and `_ext,py` a bit.

I agree that arcade games are being used a lot but that's just a reflection of what I found while doing the research for those games. It makes sense: most arcade games are pretty simple compared to more modern games, which explains why there's so many clones of them. People think "hey that should be pretty simple to code" and they do it for fun, for nostalgia or as a learning project. The result is many clones as we see here.

I agree that some of those games would be better labeled with other tags - artillery games are a good example, as you mention. I don't think though that TBS is a good alternative here. Technically there's some strategy to these games but it isn't what most people think of when they say TBS - these would be vastly more complex games like Batttle for Wesnoth or Civilization. I guess in the end it's a matter of opinion but I suggest "artillery" as a new genre label since there should be enough of them to warrant its inclusion.

File: games/s.yaml (hunk 18 above)
Comment (tukkek) [inline games/s.yaml]: oops :) and that was a pretty easy one too!

**Comment by tukkek** [comment]
I activated the "allow edits" option on this pull request but I'm not sure how to actually edit the commits.

File: games/s.yaml (hunk 15 above)
Comment (cxong) [inline games/s.yaml]: My bad; I guess I was thinking of the 2004 version. I haven't played the 1987 version but it looks like the land combat is real time in that one.

**Comment by tukkek** [comment]
You should push `schema.yaml` too.

**Comment by tukkek** [comment]
Dismiss what I said before: I misunderstood what was happening concerning the new commits. You're using the wrong syntax, it should be `genre: [arcade, platform]`

**Comment by tukkek** [comment]
Nevermind again, the pull request interface is just kinda confusing and I'm new to it. I'll try to figure out what's wrong here.

**Comment by tukkek** [comment]
Cool, errors solved. I'll make the remaining changes suggested by @cxong now that I figured out how to edit the files via GH and it should be good to go.

How to call the "platformer" genre is a good question. My first approach was either "platformer" or "platforming" but I've seen it referred a ton of times as "platform games" on Wikipedia and it's the shortest one to write so I went with that. To be honest I don't think anyone will be confused by it instead of "platforming" or other usage so I think it's alright.

**Comment by tukkek** [comment]
With the exception of Sid Meyer's Pirates I think everything is good to go now. I suggest we open a new issue after this pull request is applied so someone who knows the game better than we can provide the correct information.

**Comment by tukkek** [comment]
Just doubled down and all the changes @cxong has pointed out have been rectified - thanks for the help @wonea :) Exception being the Pirates! game as mentioned before.

**Comment by nikuda** [comment]
Thanks @tukkek. Looks good.

I would suggest making more use of the subgenre key. I don't think `artillery` genre deserves to be on the top level – see https://en.wikipedia.org/wiki/Artillery_game

**Comment by tukkek** [comment]
As I've said before I think the list isn't big enough to need the subgenres yet. It would pollute the page with too many tags while currently just having an artillery tag (for example) isn't really compromising the quality of the tag system. Maybe by the time there's too many games in a single tag and they can't be further divided (like strategy being divided into tactics too further down the road) the subgenres will be more useful. As-is I don't think they'd really be beneficial in practice.

Of course that's just my opinion and how the other folk here went around making adjustments. Feel free to edit this pull request (or the files after the pull is committed) if you think you have a strong argument for doing so!

**Comment by wonea** [comment]
Thanks for this! I'll merge.

Personally I don't feel the need for subgenres. Keep this website within the scope of what it is called. There plenty of open source clones out there which this site doesn't cover. I think the efforts should be concentrated in finding & adding those games (see wiki). I don't think we should add much more metadata to items, because this increases the time spent to add new items. I.e. keep the bar low.

**Comment by tukkek** [comment]
Agreed and this argument is relevant to #398 as well since you establish the focus of the project and that there's more work to be done on this front instead of trying to expand its ambitions. Thank you for the merge!

**Comment by piranha** [comment]
Woooow! That's really cool!

@wonea what do you think if we add more people as contributors. I'm thinking that this can only do good since I'm slacking off on my duties as maintainer, and @cxong, @nikuda and @tukkek are doing really good job. :-)

**Comment by wonea** [comment]
@piranha yes, more contributors is a good thing - we can see where this site grows. Maybe we should have a simple aim or prerogative? I.e. helping programmers find game projects.

**Comment by piranha** [comment]
All invited. :)

**Comment by piranha** [comment]
I still think current process with pull requests and somebody reviewing them is good though! Gets more eyes on changes. :)

**Comment by tukkek** [comment]
@piranha - I'd be happy to become a contributor. I would've added #410 already since I saw it a little while ago :D

**Comment by tukkek** [comment]
Nevermind, just saw the email and I'm in - I'm honored to be part of the team! You guys should write a small "developer info" on the wiki with the current rules (if any) and scope of the project, I think, especially if you're planning on inviting more people soon...

**Comment by nikuda** [comment]
>developer info

I think the usual thing of follow established project style/practices rather than making your own applies.
